metricbeat/helper/server/http/http.go

```diff
@@ -73,7 +74,10 @@ func getDefaultHttpServer(mb mb.BaseMetricSet) (*HttpServer, error) {
 	}
 
 	httpServer := &http.Server{
-		Addr: net.JoinHostPort(config.Host, strconv.Itoa(int(config.Port))),
+		Addr:         net.JoinHostPort(config.Host, strconv.Itoa(int(config.Port))),
+		ReadTimeout:  5 * time.Second,
```

Comment:
This timeout is the maximum time to read the entire request, including the body. If we receive big requests we could be hitting this limit. We can also use `ReadHeaderTimeout` with a lower value if we want to avoid getting stucked on empty connections.